File: FurnitureMoverApp/src/app/furniture.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Furniture } from './models/Furniture';

@Injectable({
  providedIn: 'root'
})
export class FurnitureService {
  // Root url for api
  private url = "http://localhost:8080/api";

  constructor(private httpCli: HttpClient) { }

  /**
   * Get all Furniture for a specified House
   * 
   * @param houseID the ID of the House
   * @returns http.get client
   */
  getFurniture(houseID: number): Observable<Furniture[]>
  {
    const options = { 
      headers: new HttpHeaders({
        'Content-Type': 'application/json'
      })
    }

    // console.log("Getting furniture from: " + `${this.url}/${houseID}/furniture`);

    return this.httpCli.get<Furniture[]>(`${this.url}/${houseID}/furniture`, options);
  }

  /**
   * Add a new Furniture to a House
   * 
   * @param houseID the House
   * @param newFurnitureName the name of the Furniture
   * @param newFurnitureSize the size of the Furniture
   * @returns http.post client
   */
  createFurniture(houseID: number, newFurnitureName: String, newFurnitureSize: Number): Observable<Furniture>
  {
    const options = { 
      headers: new HttpHeaders({
        'Content-Type': 'application/json'
      })
    }

    const body = `{
      "furnitureName": "${newFurnitureName}",
      "furnitureSize": ${newFurnitureSize}
    }`;
    
    return this.httpCli.post<Furniture>(`${this.url}/${houseID}/furniture/addFurniture`, body, options);
  }

  /**
   * Update a Furniture
   * 
   * @param houseID The house
   * @param furnitureID The furniture
   * @param newFurnitureName An updated name
   * @param newFurnitureSize An updated size
   * @returns http.put client
   */
  updateFurniture(houseID: number, furnitureID: number, newFurnitureName: String, newFurnitureSize: Number)
  {
    const options = { 
      headers: new HttpHeaders({
        'Content-Type': 'application/json'
      })
    }

    const body = `{
      "furnitureName": "${newFurnitureName}",
      "furnitureSize": ${newFurnitureSize}
    }`;
    
    return this.httpCli.put<Furniture>(`${this.url}/${houseID}/furniture/${furnitureID}/updateFurniture`, body, options);
  }

  /**
   * Delete a Furniture
   * 
   * @param houseID The house
   * @param furnitureID The furniture
   * @returns http.delete client
   */
  deleteFurniture(houseID : number, furnitureID : number)
  {
    const options = { 
      headers: new HttpHeaders({
        'Content-Type': 'application/json'
      })
    }

    console.log(`DELETING FURNITURE ${furnitureID} IN HOUSE ${houseID}`);
    console.log(`URL: ${this.url}/${houseID}/furniture/${furnitureID}/deleteFurniture`);

    return this.httpCli.delete<Furniture>(`${this.url}/${houseID}/furniture/${furnitureID}/deleteFurniture`, options);
  }
}

File: src/main/java/com/example/service/FurnitureService.java
package com.example.service;

import com.example.model.Furniture;
import com.example.model.House;

import java.util.List;

/**
 * Interface for a Furniture Service
 *
 * - Insert new Furniture to a House
 * - Get all Furniture for a House
 * - Get a Furniture by ID
 * - Update Furniture name
 * - Update Furniture size
 * - Delete Furniture
 */
public interface FurnitureService {
    /**
     * Insert a new Furniture into a specified House
     *
     * @param house the House
     * @param furniture the Furniture
     * @return the added Furniture
     */
    Furniture insertFurniture(House house, Furniture furniture);

    /**
     * Get all the Furniture for a specific House
     *
     * @param houseID id of the House to check
     * @return ArrayList of all Furniture in a House
     */
    List<Furniture> getAllFurniture(int houseID);

    /**
     * Get a Furniture by it's ID
     *
     * @param furnitureID the ID of the Furniture
     * @return the Furniture
     */
    Furniture getFurnitureByID(int furnitureID);

    /**
     * Update a Furniture with a new name and size
     *
     * @param furniture the furniture to update
     * @param newName the new name
     * @param newSize the new size
     * @return the updated Furniture
     */
    Furniture updateFurniture(Furniture furniture, String newName, double newSize);

    /**
     * Remove a Furniture
     *
     * @param furnitureID the Furniture to remove
     */
    void deleteFurniture(int furnitureID);

}
